ODIN_II/SRC/partial_map.cpp

```diff
@@ -221,9 +221,12 @@ void partial_map_node(nnode_t* node, short traverse_number, netlist_t* netlist)
             instantiate_multi_port_mux(node, traverse_number, netlist);
             break;
         case MULTIPLY: {
-            int mult_size = std::max<int>(node->input_port_sizes[0], node->input_port_sizes[1]);
-            if (hard_multipliers && mult_size >= min_mult) {
-                instantiate_hard_multiplier(node, traverse_number, netlist);
+            if (mixer->hardenable(node)) {
```

Comment:
This can be changed to 

```
if (mixer->enabled(node)) {
  mixer->note_candidate_node(node);
} else if(mixer->hardenable(node) {
  mixer->instantiate_hard(node, traverse_number, netlist);
} else {
  mixer->instantiate_soft(node, traverse_number, netlist);
}
```

Reply:
@jeanlego thanks for the comment. But what's this -> in this context? We don't have a node hierarchy, and optimization passes care about all the nodes, not individual nodes?

Reply:
Are you suggesting mixer should have a partial map function that should be called from here?

Reply:
>Are you suggesting mixer should have a partial map function that should be called from here?

correct, sorry if I wasn't too clear about that.

Reply:
Please check out the proposed solution